sklearn/gaussian_process/tests/test_kernels.py

```diff
@@ -89,39 +91,68 @@ def test_kernel_theta():
 
         # Check that values returned in theta are consistent with
         # hyperparameter values (being their logarithms)
-        for i, hyperparameter in enumerate(kernel.hyperparameters):
-            assert_equal(theta[i],
-                         np.log(getattr(kernel, hyperparameter.name)))
+        i = 0
```

Comment:
Thanks that you fixed this problem. I was suprised that it was not an issue so far because the anisotropic RBF/Matern also violate the condition. But this is because the test skips the non-basic kernels at the beginning. Could you check if you can now also remove the lines 74-76:
```         
if isinstance(kernel, KernelOperator) \
     or isinstance(kernel, Exponentiation):  # skip non-basic kernels
      continue
```